job-service-db/src/main/resources/procedures/cancelJob.sql

```diff
@@ -1,5 +1,5 @@
 --
--- Copyright 2016-2020 Micro Focus or one of its affiliates.
+-- Copyright 2015-2018 Micro Focus or one of its affiliates.
```

Comment:
I thought the procedures were safe to be changed?

Reply:
I thought so too, the documentation was vague. After a look at their source code they're explicitly including the sql files into the checksum.

Reply:
This seems worrying to me.  I don't understand how it's never been a problem before or what it means in future.  Should these procedures never be touched again?  Come and talk to me about this sometime tomorrow.

Reply:
Now that I understand the problem more, I can probably back out this part of the change and not exclude the procedures from the license check. This was all caused by the the 3.1.0 changelog which didn't have `runOnChange=true` set on the `changeSets` which modified stored procedures and so it's checksum validation would fail whenever a procedure was changed further down the line (e.g. the label changes).

I think the issue is our approach. Either we should:

-  leave the stored procedures as static and create separate, versioned sql files and apply them in `changeSets` sequentially
Or
-  Modify the stored procedures as we've done, and have a single `changeSet` tag for each procedure that has `runOnChange=true` so the latest version of the procedure is apply every time.

Currently we have a mix where we've kept modifying the stored procedure files and adding a `changeSet` every time we do. That only works if they're all flagged to run on change which wasn't the case in the 3.1.0 change log.

Reply:
Ok if those are the options then it sounds like the second one is the simpler from an ongoing maintenance perspective.

Reply:
I assume you mean this going forward and not saying to remove the duplicate `changeSets` from the current change logs? That would cause a whole load more upgrade issues where the removed `changeSets` have already been applied etc.